python/ucxx/ucxx/_lib_async/application_context.py

```diff
@@ -57,7 +57,12 @@ def __init__(
         self.enable_delayed_submission = enable_delayed_submission
         self.enable_python_future = enable_python_future
 
-        self.exchange_peer_info_timeout = exchange_peer_info_timeout
+        if "UCXPY_CONNECT_TIMEOUT" in os.environ:
+            self.connect_timeout = float(os.environ["UCXPY_CONNECT_TIMEOUT"])
+        elif connect_timeout is not None:
+            self.connect_timeout = connect_timeout
+        else:
+            self.connect_timeout = 10.0
```

Comment:
```suggestion
        if connect_timeout is None:
            self.connect_timeout = float(os.environ.get("UCXPY_CONNECT_TIMEOUT", 5))
        else:
            self.connect_timeout = connect_timeout
```

Reply:
With `None`, we can simplify this now, plus we keep the same order as in the rest of UCXX, priority to the explicit argument, then to the environment variable.